plugins/cost-insights/src/components/CostInsightsPage/CostInsightsPageRoot.tsx

```diff
@@ -22,20 +22,23 @@ import { GroupsProvider } from '../../hooks/useGroups';
 import { CurrencyProvider } from '../../hooks/useCurrency';
 import { ScrollProvider } from '../../hooks/useScroll';
 import { ConfigProvider } from '../../hooks/useConfig';
+import { BillingDateProvider } from '../../hooks/useLastCompleteBillingDate';
 import { CostInsightsThemeProvider } from './CostInsightsThemeProvider';
 
 const CostInsightsPageRoot = () => (
   <CostInsightsThemeProvider>
     <ConfigProvider>
       <LoadingProvider>
         <GroupsProvider>
-          <FilterProvider>
-            <ScrollProvider>
-              <CurrencyProvider>
-                <CostInsightsPage />
-              </CurrencyProvider>
-            </ScrollProvider>
-          </FilterProvider>
+          <BillingDateProvider>
```

Comment:
Fetching groups and billing data could be parallelized but they're being executed serially due to this waterfall.

Might be an optimization we consider for the future?